modules/apps/commerce/commerce-account-service/src/main/java/com/liferay/commerce/account/internal/upgrade/v3_0_0/CommerceAccountUpgradeProcess.java

```diff
@@ -0,0 +1,171 @@
+import com.liferay.portal.kernel.model.WorkflowDefinitionLink;
+import com.liferay.portal.kernel.model.WorkflowInstanceLink;
+import com.liferay.portal.kernel.service.ClassNameLocalServiceUtil;
+import com.liferay.portal.kernel.service.GroupLocalServiceUtil;
+import com.liferay.portal.kernel.service.ResourceLocalServiceUtil;
+import com.liferay.portal.kernel.service.WorkflowDefinitionLinkLocalServiceUtil;
+import com.liferay.portal.kernel.service.WorkflowInstanceLinkLocalServiceUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+import java.util.function.UnaryOperator;
+
+/**
+ * @author Drew Brokke
+ */
+public class CommerceAccountUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		String selectCommerceAccountSQL =
+			"select * from CommerceAccount order by commerceAccountId asc";
+
+		String deleteCommerceAccountSQL =
+			"delete from CommerceAccount where commerceAccountId = ?";
+
+		try (Statement selectStatement = connection.createStatement();
+			PreparedStatement deleteStatement = connection.prepareStatement(
+				deleteCommerceAccountSQL)) {
+
+			ResultSet rs = selectStatement.executeQuery(
+				selectCommerceAccountSQL);
+
+			while (rs.next()) {
+				long accountEntryId = rs.getLong("commerceAccountId");
+
+				AccountEntry accountEntry =
+					AccountEntryLocalServiceUtil.createAccountEntry(
+						accountEntryId);
+
+				long companyId = rs.getLong("companyId");
+
+				accountEntry.setCompanyId(companyId);
+
+				long userId = rs.getLong("userId");
+
+				accountEntry.setUserId(userId);
+
+				accountEntry.setUserName(rs.getString("userName"));
+				accountEntry.setCreateDate(rs.getTimestamp("createDate"));
+				accountEntry.setModifiedDate(rs.getTimestamp("modifiedDate"));
+				accountEntry.setDefaultBillingAddressId(
+					rs.getLong("defaultBillingAddressId"));
+				accountEntry.setDefaultShippingAddressId(
+					rs.getLong("defaultShippingAddressId"));
+				accountEntry.setParentAccountEntryId(
+					rs.getLong("parentCommerceAccountId"));
+				accountEntry.setEmailAddress(rs.getString("email"));
+				accountEntry.setLogoId(rs.getLong("logoId"));
+				accountEntry.setName(rs.getString("name"));
+				accountEntry.setTaxIdNumber(rs.getString("taxId"));
+				accountEntry.setType(
+					CommerceAccountImpl.toAccountEntryType(rs.getInt("type_")));
+				accountEntry.setStatus(
+					CommerceAccountImpl.toAccountEntryStatus(
+						rs.getBoolean("active_")));
+
+				accountEntry.setExternalReferenceCode(
+					rs.getString("externalReferenceCode"));
+
+				AccountEntryLocalServiceUtil.addAccountEntry(accountEntry);
+
+				ResourceLocalServiceUtil.addResources(
+					companyId, 0, userId, AccountEntry.class.getName(),
+					accountEntryId, false, false, false);
+
+				deleteStatement.setLong(1, accountEntryId);
+
+				deleteStatement.executeUpdate();
```

Comment:
I don't think we can drop the table yet since we're not yet removing the service. I think as long as the service exists the table will be re-added, right?

Reply:
that's right, then we can keep it since we still have it in the service